pynicotine/gtkgui/frame.py

```diff
@@ -1468,14 +1468,45 @@ def on_update_user_info(self, *_args):
     def on_get_user_info(self, *_args):
         self.userinfo.on_get_user_info()
 
-    """ Browse Shares """
+    """ Shares """
 
     def on_get_shares(self, *_args):
         self.userbrowse.on_get_shares()
 
     def on_load_from_disk(self, *_args):
         self.userbrowse.on_load_from_disk()
 
+    def shares_unavailable_response(self, _dialog, response_id, _data):
+
+        if response_id == 2:  # 'Retry'
+            self.core.shares.rescan_shares()
+
+        elif response_id == 3:  # 'Force Rescan'
+            self.core.shares.rescan_shares(force=True)
+
+    def shares_unavailable(self, shares):
+
+        shares_list_message = ""
+
+        for virtual_name, folder_path in shares:
+            shares_list_message += "• %s (%s)\n" % (folder_path, virtual_name)
+
+        def create_dialog():
+            OptionDialog(
+                parent=self.window,
+                title=_("Shares Not Available"),
+                message=_("The following shares are not available. Verify that external disks are mounted and "
+                          "folder permissions are correct."),
+                long_message=shares_list_message,
+                first_button=_("_Cancel"),
+                second_button=_("_Retry"),
+                third_button=_("_Force Rescan"),
```

Comment:
The Force Rescan button can and should be hidden if there is nothing available to scan, because there is no point in offering to completely wipe out the index, since the primary purpose of all this is to prevent inadvertent data loss.

`third_button=_("_Force Rescan") if shares else "",`

If the user intends to zero their index on purpose, then it will be possible for the rescan command to be run with the force argument in any case.

Reply:
I'm not quite sure what you mean. The warning (in my branch) only appears if you have shared something, and at least one of the shared folders (with a virtual name) are unavailable.

If you're talking about showing a warning if a user hasn't shared any folders (0 folders, 0 files), I don't think it belongs in this dialog.

Reply:
> Since the dialog is unaware of the total number of configured shares, then it's understandable that this complexity is not desired at the moment. This current implementation seems to satisfy the requirement nicely.

It could be added, but I'm not sure how common it would be for all shares to be unavailable at the same time. Leaning towards not adding this for now.